Upgrade ConfigReader to ConfigConvert for http4s Uri

LMnet opened the pull request:
Minor improvement: upgraded `ConfigReader` to `ConfigConvert` for http4s `Uri`.

LMnet committed:
Upgrade ConfigReader to ConfigConvert for http4s Uri

leifwickland reviewed (approved):
Thanks, @LMnet.

@ruippeixotog & @jcazevedo This change is not binary compat. Do you think that should have any impact on how we version the release that includes this?

LMnet commented:
@leifwickland oh, I completely forgot about that.

I can provide `ConfigWriter` and `ConfigConvert` without touching the `ConfigReader`, so it will binary compatible.

ruippeixotog commented on modules/http4s/src/main/scala/pureconfig/module/http4s/package.scala:
```diff
@@ -1,8 +1,8 @@
 import org.http4s.Uri
-import pureconfig.ConfigReader
 import pureconfig.error.CannotConvert
+import pureconfig.{ConfigReader, ConfigWriter}
```
The build is failing because this line doesn't conform to the Scalariform rule (running `sbt compile` should fix this automatically). Could you please fix this?

ruippeixotog reviewed (changes requested):
> @ruippeixotog & @jcazevedo This change is not binary compat. Do you think that should have any impact on how we version the release that includes this?

We don't usually have this requirement in our core library and we are even less strict on modules, so I think this is totally fine.

jcazevedo reviewed (changes requested):
@LMnet: Thanks for your PR! Can you please add a test for round-trip reading/writing of `Uri`?

LMnet commented:
> @LMnet: Thanks for your PR! Can you please add a test for round-trip reading/writing of `Uri`?

@jcazevedo Done. I added a test and fixed the formatting.

jcazevedo reviewed (approved):
@LMnet: Thanks for this!
